File: backend/interview-service/src/main/java/com/careerflow/interviewservice/interview/model/Interview.java
package com.careerflow.interviewservice.interview.model;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.Id;
import jakarta.persistence.PrePersist;
import jakarta.persistence.PreUpdate;
import jakarta.persistence.Table;
import jakarta.persistence.Version;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.Instant;
import java.util.UUID;

@Entity
@Table(name = "interviews")
@Getter
@Setter
@NoArgsConstructor
public class Interview {

    @Id
    private UUID id;

    @Column(name = "user_id", nullable = false)
    private String userId;

    @Column(name = "application_id", nullable = false)
    private UUID applicationId;

    @Column(name = "round_number", nullable = false)
    private int roundNumber;

    @Enumerated(EnumType.STRING)
    @Column(name = "round_type", nullable = false)
    private RoundType roundType;

    @Column(name = "title")
    private String title;

    @Enumerated(EnumType.STRING)
    @Column(name = "mode", nullable = false)
    private InterviewMode mode;

    @Column(name = "scheduled_at", nullable = false)
    private Instant scheduledAt;

    @Column(name = "duration_minutes")
    private Integer durationMinutes;

    @Column(name = "meeting_link")
    private String meetingLink;

    @Column(name = "location")
    private String location;

    @Column(name = "interviewer_names")
    private String interviewerNames;

    @Enumerated(EnumType.STRING)
    @Column(name = "status", nullable = false)
    private InterviewStatus status;

    @Enumerated(EnumType.STRING)
    @Column(name = "outcome", nullable = false)
    private InterviewOutcome outcome;

    @Column(name = "notes")
    private String notes;

    @Column(name = "created_at", nullable = false)
    private Instant createdAt;

    @Column(name = "updated_at", nullable = false)
    private Instant updatedAt;

    @Version
    @Column(name = "version", nullable = false)
    private Long version;

    @PrePersist
    void onCreate() {
        if (id == null) {
            id = UUID.randomUUID();
        }
        Instant now = Instant.now();
        createdAt = now;
        updatedAt = now;
        if (status == null) {
            status = InterviewStatus.SCHEDULED;
        }
        if (outcome == null) {
            outcome = InterviewOutcome.PENDING;
        }
    }

    @PreUpdate
    void onUpdate() {
        updatedAt = Instant.now();
    }
}

File: frontend/src/types/interview.ts
export type RoundType =
  | 'PHONE_SCREEN'
  | 'TECHNICAL'
  | 'SYSTEM_DESIGN'
  | 'BEHAVIORAL'
  | 'HIRING_MANAGER'
  | 'HR'
  | 'ONSITE'

export type InterviewMode = 'REMOTE' | 'ONSITE' | 'PHONE'

export type InterviewStatus =
  | 'SCHEDULED'
  | 'COMPLETED'
  | 'CANCELLED'
  | 'RESCHEDULED'
  | 'NO_SHOW'

export type InterviewOutcome = 'PENDING' | 'PASSED' | 'FAILED' | 'PENDING_DECISION'

export interface Interview {
  id: string
  applicationId: string
  roundNumber: number
  roundType: RoundType
  title: string | null
  mode: InterviewMode
  scheduledAt: string
  durationMinutes: number | null
  meetingLink: string | null
  location: string | null
  interviewerNames: string | null
  status: InterviewStatus
  outcome: InterviewOutcome
  notes: string | null
  createdAt: string
  updatedAt: string
  version: number
}

export interface InterviewStatsResponse {
  totalInterviews: number
  activeInterviews: number
  upcomingInterviews: number
  completedInterviews: number
  interviewsByStatus: Partial<Record<InterviewStatus, number>>
}

export interface CreateInterviewRequest {
  applicationId: string
  roundType: RoundType
  title?: string | null
  mode: InterviewMode
  scheduledAt: string
  durationMinutes?: number | null
  meetingLink?: string | null
  location?: string | null
  interviewerNames?: string | null
  notes?: string | null
}

export interface UpsertRetrospectiveRequest {
  whatWentWell?: string | null
  whatToImprove?: string | null
  questionsAsked?: string | null
  selfRating?: number | null
  followUpActions?: string | null
}

export interface Retrospective {
  id: string
  interviewId: string
  whatWentWell: string | null
  whatToImprove: string | null
  questionsAsked: string | null
  selfRating: number | null
  followUpActions: string | null
  createdAt: string
  updatedAt: string
}

export interface InterviewListParams {
  applicationId?: string
  status?: InterviewStatus
  page?: number
  size?: number
}

export interface InterviewListResponse {
  content: Interview[]
  totalElements: number
  totalPages: number
  number: number
  size: number
}
